freqtrade/optimize/hyperopt_interface.py

```diff
@@ -76,30 +81,102 @@ def generate_roi_table(params: Dict) -> Dict[int, float]:
+            'roi_p2_min': 0.01 * roi_p_scale * roi_p_alpha,
+            'roi_p2_max': 0.07 * roi_p_scale * roi_p_alpha,
+            'roi_p3_min': 0.01 * roi_p_scale * roi_p_alpha,
+            'roi_p3_max': 0.20 * roi_p_scale * roi_p_alpha,
+        }
+        logger.debug(f"Using roi space limits: {roi_limits}")
+        p = {
+            'roi_t1': roi_limits['roi_t1_min'],
+            'roi_t2': roi_limits['roi_t2_min'],
+            'roi_t3': roi_limits['roi_t3_min'],
+            'roi_p1': roi_limits['roi_p1_min'],
+            'roi_p2': roi_limits['roi_p2_min'],
+            'roi_p3': roi_limits['roi_p3_min'],
+        }
+        logger.info(f"Min roi table: {IHyperOpt.generate_roi_table(p)}")
+        p = {
+            'roi_t1': roi_limits['roi_t1_max'],
+            'roi_t2': roi_limits['roi_t2_max'],
+            'roi_t3': roi_limits['roi_t3_max'],
+            'roi_p1': roi_limits['roi_p1_max'],
+            'roi_p2': roi_limits['roi_p2_max'],
+            'roi_p3': roi_limits['roi_p3_max'],
+        }
+        logger.info(f"Max roi table: {IHyperOpt.generate_roi_table(p)}")
 
-        Defines range of stoploss values to search.
-        You may override it in your custom Hyperopt class.
-        """
         return [
-            Real(-0.5, -0.02, name='stoploss'),
+            Integer(roi_limits['roi_t1_min'], roi_limits['roi_t1_max'], name='roi_t1'),
+            Integer(roi_limits['roi_t2_min'], roi_limits['roi_t2_max'], name='roi_t2'),
+            Integer(roi_limits['roi_t3_min'], roi_limits['roi_t3_max'], name='roi_t3'),
+            Real(roi_limits['roi_p1_min'], roi_limits['roi_p1_max'], name='roi_p1'),
+            Real(roi_limits['roi_p2_min'], roi_limits['roi_p2_max'], name='roi_p2'),
+            Real(roi_limits['roi_p3_min'], roi_limits['roi_p3_max'], name='roi_p3'),
         ]
 
     @staticmethod
-    def roi_space() -> List[Dimension]:
+    def stoploss_space() -> List[Dimension]:
         """
-        Create a ROI space.
+        Create a stoploss space.
 
-        Defines values to search for each ROI steps.
+        Defines range of stoploss values to search.
         You may override it in your custom Hyperopt class.
         """
         return [
-            Integer(10, 120, name='roi_t1'),
-            Integer(10, 60, name='roi_t2'),
-            Integer(10, 40, name='roi_t3'),
-            Real(0.01, 0.04, name='roi_p1'),
-            Real(0.01, 0.07, name='roi_p2'),
-            Real(0.01, 0.20, name='roi_p3'),
+            Real(-0.5, -0.02, name='stoploss'),
         ]
+
+    # This is needed for proper unpickling the class attribute ticker_interval
+    # which is set to the actual value by the resolver.
+    # Why do I still need such shamanic mantras in modern python?
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state['ticker_interval'] = self.ticker_interval
+        return state
+
+    def __setstate__(self, state):
```

Comment:
i assume this is needed due to the change in hyperopt_resolver?

i'm not sure i don't prefer the previous method ... which accesses the `__class__` attribute directly.
Or did that cause problems with this PR .. ?

Reply:
Yes, without this shamanism the ticker_interval is not unpickled in the joblib worker processes. I just wonder why it's impossible to simply pickle/unpickle all attributes currently existing in the class or instance without forcing me to write such the isoteric mantras.

Btw, python 3.8 as far as I read will allow brand new pickling mechanics...

Reply:
When ticker_interval is assigned as `__class__` attribute (as it was previously implemented in the resolver), this function in the IHyperOpt does not see it, because it's attribute of the derived class. And the ticker_interval class attribute declared here above remains unset.

Reply:
hm i see, yes we addet it to the child-class... (which wasn't a problem when i added this) :D 

Pickle itself has a few problems - the biggest introduced by our resolver mechanisms - since it needs to "refind" classes (`wrap_non_picklable_objects()` fixes this for the biggest part). If it doesn't - it freaks.

The problem with "the next version has this better" is that it'll take time for everyone to upgrade, and forcing 3.8 right away will be a pain to support.

assuming most people use their OS python if possible, it'll be months until all distributions have upgraded...

Reply:
> The problem with "the next version has this better" is that it'll take time for everyone to upgrade, and forcing 3.8 right away will be a pain to support.
> 
> assuming most people use their OS python if possible, it'll be months until all distributions have upgraded...

Sure. I'm not proposing to move to python 3.8, not at all. I only mentioned they completely reworked pickling in 3.8, as far as I know, this may be easier to use in the future...